src/sys/windows/selector.rs

```diff
@@ -351,42 +354,55 @@ impl Selector {
 
 #[derive(Debug)]
 pub struct SelectorInner {
+    lock: Mutex<()>,
     cp: CompletionPort,
-    active_poll_count: AtomicUsize,
-    update_queue: Mutex<VecDeque<Arc<Mutex<SockState>>>>,
+    active_poll_count: RefCell<usize>,
+    update_queue: RefCell<VecDeque<Arc<Mutex<SockState>>>>,
     afd_group: Mutex<Vec<Arc<Afd>>>,
 }
 
+// We have ensured thread safety by introducing lock manually.
+unsafe impl Sync for SelectorInner {}
```

Comment:
You can't implement `Sync` when using `RefCell`, you must use `UnsafeCell` and ensure that only a single thread can access it any given time.